File: CPP/ByRound/2111/2111_A_Energy_Crystals.cpp
// Problem: CF 2111 A - Energy Crystals
// https://codeforces.com/contest/2111/problem/A

#include <cstdio>
int main() {
  long t;
  scanf("%ld", &t);
  while (t--) {
    long x;
    scanf("%ld", &x);
    long g(1), cnt(0); // g represents the maximum energy level that can be achieved with 'cnt' steps, starting from 0
    while (g < x) {    // incrementally build up the maximum possible value using the recurrence relation g = 2*g + 1
      g = 2 * g + 1;
      ++cnt;
    }
    printf("%ld\n", 2 * cnt + 3); // the number of operations needed is derived from the structure of the optimal charging strategy
  }
}
/*
Problem: A. Energy Crystals

Algorithm/Techniques: Mathematical analysis and binary representation concepts to simulate optimal charging process.

Time Complexity: O(log x) per test case — The loop runs until g >= x, and each step doubles g and adds 1, so it's logarithmic in x.
Space Complexity: O(1) — Only a constant amount of extra space is used.

The problem involves optimally charging three crystals under synchronization constraints. 
Each operation must maintain a valid energy ratio between crystals (ai ≥ floor(aj / 2)) for all pairs i,j.
The solution uses a recursive structure that mirrors the binary representation and optimal growth pattern.
*/

// https://github.com/VaHiX/codeForces/

File: Python/ByRound/2111/2111_A_Energy_Crystals.py
# Problem: CF 2111 A - Energy Crystals
# https://codeforces.com/contest/2111/problem/A

# ==============================================================================
# Problem: Energy Crystals
# Algorithm/Technique: Mathematical Analysis, Binary Representation
# Time Complexity: O(1) per test case
# Space Complexity: O(1)
#
# The problem involves charging three energy crystals to a target level x.
# The constraint is that after each action, for any two crystals i and j:
#   a[i] >= floor(a[j]/2)
# This means the energy levels must be such that no crystal has less than half
# the energy of another (rounded down).
#
# Key Insight:
# The optimal way to charge all crystals is through a sequence where we build up
# the energy in a balanced fashion. The minimum number of operations can be derived
# using bit manipulation and binary representation.
#
# For x > 0, the solution uses:
# len(bin(x)) * 2 - 3
# This expression arises from analyzing how many steps are needed to reach balance
# through optimal charging strategies in a binary tree-like structure.
# ==============================================================================

for x in [*open(0)][1:]:  # Read input lines starting from second line (skip t)
    print(len(bin(int(x))) * 2 - 3)  # Compute and output the result using bit length


# https://github.com/VaHiX/codeForces/